code/modules/power/stirling.dm

```diff
@@ -0,0 +1,218 @@
+			spark_at(src, cardinal_only = TRUE)
+
+	air1.add_thermal_energy(air1_dq)
+	air2.add_thermal_energy(air2_dq)
+	
+	generate_power(power_generated)
+	lastgen = power_generated
+	genlev = max(0, min(round(4*power_generated / max_power), 4))
+
+	update_icon()
+	update_sound()
+	update_networks()
+
+/obj/machinery/atmospherics/binary/stirling/examine(mob/user, distance)
+	. = ..()
+	if(distance > 1)
+		return
+
+	to_chat(user, "\The [src] is generating [round(lastgen/1000, 0.1)] kW")
+	if(!inserted_cylinder)
+		to_chat(user, "There is no piston cylinder inserted into \the [src].")
+
+/obj/machinery/atmospherics/binary/stirling/attackby(var/obj/item/W, var/mob/user)
+	if((istype(W, /obj/item/tank/stirling)))
+		if(inserted_cylinder)
+			return
+		if(!user.unEquip(W, src))
+			return
+		to_chat(user, SPAN_NOTICE("You insert \the [W] into \the [src]."))
+		inserted_cylinder = W
+		update_icon()
+		return TRUE
+	
+	if(isCrowbar(W) && inserted_cylinder)
+		inserted_cylinder.dropInto(get_turf(src))
+		to_chat(user, SPAN_NOTICE("You remove \the [inserted_cylinder] from \the [src]."))
+		inserted_cylinder = null
+		stop_engine()
+		return TRUE
+
+	if(panel_open && isWrench(W))
+		var/target_frequency = input(user, "Enter the cycle frequency you would like \the [src] to operate at ([MAX_FREQUENCY/4] - [MAX_FREQUENCY] Hz)", "Stirling Frequency", cycle_frequency) as num | null
+		if(!CanPhysicallyInteract(user) || !target_frequency)
+			return
+		cycle_frequency = round(clamp(target_frequency, MAX_FREQUENCY/4, MAX_FREQUENCY))
+		to_chat(usr, SPAN_NOTICE("You adjust \the [src] to operate at a frequency of [cycle_frequency] Hz."))
+		return TRUE
+
+	. = ..()
+
+/obj/machinery/atmospherics/binary/stirling/attack_hand(mob/user)
+	if(!(stat & BROKEN) && use_power != POWER_USE_ACTIVE)
+		if(!inserted_cylinder)
+			to_chat(user, SPAN_WARNING("You must insert a stirling piston cylinder into \the [src] before you can start it!"))
+			return
+		to_chat(user, "You start trying to manually rev up \the [src].")
+		if(do_after(user, 2 SECONDS, src) && use_power != POWER_USE_ACTIVE && inserted_cylinder && !(stat & BROKEN))
+			visible_message("[user] pulls on the starting cord of \the [src], revving it up!", "You pull on the starting cord of \the [src], revving it up!")
+			playsound(src.loc, 'sound/machines/engine.ogg', 35, 1)
+			update_use_power(POWER_USE_ACTIVE)
+		return
+	. = ..()
+
+/obj/machinery/atmospherics/binary/stirling/on_update_icon()
+	cut_overlays()
+	if (stat & (BROKEN))
+		return
+	if (genlev != 0)
+		add_overlay(emissive_overlay('icons/obj/power.dmi', "stirling-op[genlev]")) 
+
+/obj/machinery/atmospherics/binary/stirling/proc/update_sound()
+	if(!sound_id)
+		sound_id = "[type]_[sequential_id(/obj/machinery/atmospherics/binary/stirling)]"
+	if(use_power == POWER_USE_ACTIVE)
+		var/volume = 10 + 15*genlev
+		if(!sound_token)
+			sound_token = play_looping_sound(src, sound_id, 'sound/machines/engine.ogg', volume = volume)
+		sound_token.SetVolume(volume)
```

Comment:
Same here maybe, unless SetVolume is a null op when the volume is the same.

Reply:
It's more or less a noop.